test/integration/jobschedulingpreference_test.go

```
@@ -0,0 +1,264 @@
+	restclient.AddUserAgent(config, "rsp-test")
+	client := clientset.NewForConfigOrDie(config)
+
+	return fixture, client
+}
+
+func jspSpecWithoutClusterWeights(parallelism, completions int32) fedschedulingv1a1.JobSchedulingPreferenceSpec {
+	return fedschedulingv1a1.JobSchedulingPreferenceSpec{
+		TotalParallelism: parallelism,
+		TotalCompletions: completions,
+	}
+}
+
+// This assumes test setup using 2 clusters
+func jspSpecWithClusterWeights(parallelism, completions, w1, w2 int32, clusters []string) fedschedulingv1a1.JobSchedulingPreferenceSpec {
+	jspSpec := jspSpecWithoutClusterWeights(parallelism, completions)
+	jspSpec.ClusterWeights = map[string]int32{
+		clusters[0]: w1,
+		clusters[1]: w2,
+	}
+
+	return jspSpec
+}
+func createJSPTestObjs(jspSpec fedschedulingv1a1.JobSchedulingPreferenceSpec, namespace string, fedClient clientset.Interface) (string, error) {
+	// TODO(irfanurrehman): should the order of this creation matter?
+	t, err := fedClient.CoreV1alpha1().FederatedJobs(namespace).Create(getFederatedJobTemplate(namespace))
+	if err != nil {
+		return "", err
+	}
+
+	jsp := &fedschedulingv1a1.JobSchedulingPreference{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      t.Name,
+			Namespace: namespace,
+		},
+		Spec: jspSpec,
+	}
+	_, err = fedClient.SchedulingV1alpha1().JobSchedulingPreferences(namespace).Create(jsp)
+	if err != nil {
+		return "", err
+	}
+
+	return t.Name, nil
+}
+
+func waitForMatchingJobPlacement(fedClient clientset.Interface, name, namespace string, expected map[string]adapters.JobSchedulingResult) error {
+	err := wait.Poll(framework.DefaultWaitInterval, wait.ForeverTestTimeout, func() (bool, error) {
+		clusterNames := []string{}
+		p, err := fedClient.CoreV1alpha1().FederatedJobPlacements(namespace).Get(name, metav1.GetOptions{})
+		clusterNames = p.Spec.ClusterNames
+		if err != nil {
+			return false, nil
```

Why is it desirable to ignore this error (other than `IsNotFound`)?  Consider logging relevant errors ato aid in debugging.

---

Fixed!